packages/client/lib/rpc/modules/engine.ts

```diff
@@ -397,38 +360,24 @@ export class Engine {
           payloadData.blockHash
         }, received: ${bufferToHex(block.hash())}`
         this.config.logger.debug(validationError)
-        const latestValidHash = await validHash(
-          toBuffer(header.parentHash),
-          this.validBlocks,
-          this.chain
-        )
+        const latestValidHash = await validHash(toBuffer(header.parentHash), this.chain)
         const response = { status: Status.INVALID_BLOCK_HASH, latestValidHash, validationError }
         this.connectionManager.lastNewPayload({ payload: params[0], response })
         return response
       }
     } catch (error) {
       const validationError = `Error verifying block during init: ${error}`
       this.config.logger.debug(validationError)
-      const latestValidHash = await validHash(
-        toBuffer(header.parentHash),
-        this.validBlocks,
-        this.chain
-      )
+      const latestValidHash = await validHash(toBuffer(header.parentHash), this.chain)
       const response = { status: Status.INVALID, latestValidHash, validationError }
       this.connectionManager.lastNewPayload({ payload: params[0], response })
       return response
     }
 
-    const vmCopy = this.vm.copy()
     const vmHead = this.chain.headers.latest!
     let blocks: Block[]
     try {
-      blocks = await recursivelyFindParents(
-        vmHead.hash(),
-        block.header.parentHash,
-        this.validBlocks,
-        this.chain
-      )
+      blocks = await recursivelyFindParents(vmHead.hash(), block.header.parentHash, this.chain)
```

Comment:
This is just a nit but can we set the type of `block` as `let block : Block` up above at line 350 so Typescript knows what it is?

Reply:
At least in VS Code, intellisense was calling `block` an `any`, though not obvious to me why.

Reply:
have added the type